mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,951 @@
+                 norm_cfg=dict(type='BN2d')):
+        super(CenterHead, self).__init__()
+
+        num_classes = [len(t['class_names']) for t in tasks]
+        self.class_names = [t['class_names'] for t in tasks]
+        self.code_weights = code_weights
+        self.weight = weight  # weight between hm loss and loc loss
+        self.dataset = dataset
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self.encode_background_as_zeros = True
+        self.use_sigmoid_score = True
+        self.in_channels = in_channels
+        self.num_classes = num_classes
+
+        self.crit = build_loss(crit)
+        self.crit_reg = build_loss(crit_reg)
+        self.bbox_coder = build_bbox_coder(bbox_coder)
+        self.loss_aux = None
+        assert dataset == 'nuscenes'
+        self.num_anchor_per_locs = [n for n in num_classes]
+        self.use_direction_classifier = False
+
+        self.bev_only = True if mode == 'bev' else False
+
+        # a shared convolution
+        self.shared_conv = nn.Sequential(
+            build_conv_layer(
+                conv_cfg,
+                in_channels,
+                share_conv_channel,
+                kernel_size=3,
+                padding=1,
+                bias=True),
+            build_norm_layer(norm_cfg, share_conv_channel)[1],
+            nn.ReLU(inplace=True))
+
+        self.tasks = nn.ModuleList()
+
+        for num_cls in num_classes:
+            heads = copy.deepcopy(common_heads)
+            if not dcn_head:
+                heads.update(dict(hm=(num_cls, num_hm_conv)))
+                self.tasks.append(
+                    SepHead(
+                        share_conv_channel,
+                        heads,
+                        init_bias=init_bias,
+                        final_kernel=3))
+            else:
+                self.tasks.append(
+                    DCNSepHead(
+                        share_conv_channel,
+                        num_cls,
+                        heads,
+                        init_bias=init_bias,
+                        final_kernel=3))
+
+    def init_weights(self):
+        pass
+
+    def forward_single(self, x):
+        """Forward function for CenterPoint.
+
+        Args:
+            x (torch.Tensor): Input feature map with the shape of
+                [B, 512, 128, 128].
+
+        Returns:
+            list[dict]: Output results for tasks.
+        """
+        ret_dicts = []
+
+        x = self.shared_conv(x)
+
+        for task in self.tasks:
+            ret_dicts.append(task(x))
+
+        return ret_dicts
+
+    def forward(self, feats):
+        """Forward pass.
+
+        Args:
+            feats (list[torch.Tensor]): Multi-level features, e.g.,
+                features produced by FPN.
+
+        Returns:
+            tuple(list[dict]): Output results for tasks.
+        """
+        return multi_apply(self.forward_single, feats)
+
+    def _sigmoid(self, x):
```

Comment:
Since this function is general, we can move it to mmdet/models/utils and name it as `clip_sigmoid` with `eps=1e-4` as additional arguments.